frontend/catalyst/api_extensions/error_mitigation.py

```diff
@@ -188,8 +190,8 @@ def __call__(self, *args, **kwargs):
         # TODO: remove the following check once #755 is completed
         if folding == Folding.RANDOM:
             raise NotImplementedError(f"Folding type {folding.value} is being developed")
-
-        results = zne_p.bind(*args_data, self.num_folds, folding=folding, jaxpr=jaxpr, fn=self.fn)
+        fn = jaxpr.eqns[0].params.get("fn")
```

Comment:
I think there should at least be an assertion encoding your assumption since we are blindly accessing the jaxpr. What assumption are you making here, that the first equation in the jaxpr is a particular primitive? (func_p or something like that?)